common/core/src/commonMain/kotlin/com/trafi/core/model/Profile.kt

```diff
@@ -26,22 +26,22 @@ import kotlinx.serialization.internal.CommonEnumSerializer
  */
 @Serializable
 data class Profile(
-    @SerialName(value = "gender") @Required val gender: Profile.Gender,
-    @SerialName(value = "ext") @Required val ext: kotlin.collections.Map<kotlin.String, kotlin.String>,
-    @SerialName(value = "firstName") val firstName: kotlin.String? = null,
-    @SerialName(value = "lastName") val lastName: kotlin.String? = null,
-    @SerialName(value = "displayName") val displayName: kotlin.String? = null,
-    @SerialName(value = "email") val email: kotlin.String? = null,
-    @SerialName(value = "address") val address: Address? = null,
-    @SerialName(value = "birthDate") val birthDate: kotlin.String? = null
+    @SerialName(value = "gender") @Required var gender: Profile.Gender,
+    @SerialName(value = "ext") @Required var ext: kotlin.collections.Map<kotlin.String, kotlin.String>,
+    @SerialName(value = "firstName") var firstName: kotlin.String? = null,
```

Comment:
Instead of making all properties mutable, what do you think about adding "with"-style extension functions in core/iosMain, e.g.

```kotlin
fun Profile.withFirstName(firstName: String): Profile = copy(firstName = firstName)
```

Hopefully [Kotlin support for value classes](https://github.com/Kotlin/KEEP/blob/master/notes/value-classes.md) is not far away, which should help us get rid of these "withers" eventually.

Reply:
We can't use binding in this case.

Reply:
Could you clarify, please? I really don't want to make these base classes mutable as then we open up the door to all sorts of bugs 😞 These aren't structs.

Reply:
This is good example, would be better if could use @dynamicMemberLookup on AppState.

```
    var firstName: String {
        get { user?.profile.firstName ?? "" }
        set { user?.profile.firstName = newValue }
    }
```

and if properties aren't mutable we must create a new class object instead of reusing current one when we want to update user name. Besides that i don't think that we want to write a bunch of extensions for single property changes and end result should look something like this which i would say looks kinda heavy for just firstName property changes:

```
    var firstName: String {
        get { user?.profile.firstName ?? "" }
        set {
            let modifiedProfile = self.user.profile?.withFirstName(firstName)
            let modifiedUser = self.user.withProfile(modifiedProfile)
            self.user = modifiedUser
        }
    }
```

Reply:
It's not just about binding.. It's just not comfortable for normal usage in swift

Reply:
should we consider generating swift structs?

Reply:
Discussed in person, let's leave it as proposed in this PR for now